torch/distributed/checkpoint/state_dict.py

```diff
@@ -653,7 +659,11 @@ def _load_optim_state_dict(
         return
 
     for optim in optimizers:
-        optim_state_dict = _split_optim_state_dict(model, optim, state_dict, info)
+        _init_optim_state(optim)
```

Comment:
`_init_optim_state` seems to update `model.parameters()` for `Adam` even though we set `grad=0` ? 

repro: `pytest test_distributed.py` P1233005758

Reply:
Fixed with https://github.com/pytorch/pytorch/pull/125708